fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfStreamContext.java

```diff
@@ -0,0 +1,183 @@
+        this.stream = empty();
+    }
+
+    /**
+     * Retrieve the underlying {@link Stream} of {@link Triple}s.
+     * @return the stream
+     */
+    public Stream<Triple> stream() {
+        return stream;
+    }
+
+    /**
+     * Add new triples to this stream
+     *
+     * @param stream a stream of Triples to add
+     * @return the RdfStreamContext for continued use
+     */
+    public RdfStreamContext concat(final Stream<Triple> stream) {
+        return new RdfStreamContext(Stream.concat(this.stream, stream)).namespaces(namespaces()).topic(topic());
+    }
+
+    /**
+     * Returns the proffered {@link Triple}s with the context of this RdfStreamContext.
+     *
+     * @param stream the stream
+     * @return proffered Triples with the context of this RDFStream
+     */
+    public RdfStreamContext withThisContext(final Stream<Triple> stream) {
+        return new RdfStreamContext(stream).namespaces(namespaces()).topic(topic());
+    }
+
+    /**
+     * Set a {@link Node} context for this stream
+     * @param node the Node
+     * @return the original RdfStreamContext
+     */
+    public RdfStreamContext topic(final Node node) {
+        this.topic = node;
+        return this;
+    }
+
+    /**
+     * Retrieve the {@link Node} context for this stream
+     * @return the topic Node
+     */
+    public Node topic() {
+        return topic;
+    }
+
+    /**
+     * Set a Map of namespace for this stream
+     * @param namespaces the namespace map
+     * @return the original RdfStreamContext
+     */
+    public RdfStreamContext namespaces(final Map<String, String> namespaces) {
+        this.namespaces.putAll(namespaces);
+        return this;
+    }
+
+    /**
+     * Set a single namespace for this stream
+     * @param prefix the namespace prefix
+     * @param uri the full URI for this namespace
+     * @return the original RdfStreamContext
+     */
+    public RdfStreamContext namespace(final String prefix, final String uri) {
+        this.namespaces.put(prefix, uri);
+        return this;
+    }
+
+    /**
+     * @return Namespaces in scope for this stream.
+     */
+    public Map<String, String> namespaces() {
+        return namespaces;
+    }
+
+    /**
+     * Set the session context for this stream
+     * @param session the Jcr Session
+     * @return the original RdfStreamContext
+     */
+    public RdfStreamContext context(final Session session) {
+        this.context = session;
+        return this;
+    }
+
+    /**
+     * Retrieve the {@link Session} context for this stream
+     * @return the Jcr Session
+     */
+    public Session context() {
+        return context;
+    }
+
+    /**
+     * A namespace-aware {@link Collector}
+     * @return a Collector for use with aggregating the Triple stream.
+     */
+    public Collector<Triple, ?, Model> toModel() {
+        final Model model = createDefaultModel();
+        model.setNsPrefixes(namespaces());
+        return Collector.of(() -> model,
```

Comment:
Very well, they will be dropped. As for why a `Collector`, that's because `Stream::collect` requires one, which fits the java8 `Stream` model better than creating our own thing.